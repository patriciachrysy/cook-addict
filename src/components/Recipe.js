/* eslint-disable jsx-a11y/media-has-caption */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { RiTimerLine } from 'react-icons/ri';
import { BsPeopleFill } from 'react-icons/bs';
import { AiFillStar } from 'react-icons/ai';
import { getRecipe } from '../Redux/recipes/recipes';
import Loader from './Loader';
import '../styles/Recipe.css';

const Recipe = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const recipesData = useSelector((state) => state.recipes);
  const { loader, recipe } = recipesData;

  useEffect(() => {
    dispatch(getRecipe(id));
  }, []);

  return (
    <div>
      { loader && <Loader /> }
      { !loader && (
        <div className="recipe-details">
          <div className="recipe-banner">
            { recipe.original_video_url && (
            <video src={recipe.original_video_url} autoPlay="true" controls="controls" />
            )}
            { !recipe.video_url && (
            <img src={recipe.thumbnail_url} alt={recipe.name} />
            )}
          </div>
          <div className="recipe-infos">
            <h1>{recipe.name}</h1>
            <div className="metrics">
              <span>
                {' '}
                <RiTimerLine />
                {' '}
                {recipe.prep_time_minutes}
              </span>
              <span>
                {' '}
                <BsPeopleFill />
                {' '}
                {recipe.num_servings}
                People
              </span>
              { recipe.user_ratings && (
              <span>
                {' '}
                <AiFillStar />
                {' '}
                {recipe.user_ratings.score}
              </span>
              )}
            </div>
            <div className="tags">
              {
              recipe.tags.map((tag) => (
                <span key={tag.id}>{tag.display_name}</span>
              ))
              }
            </div>
          </div>
          <div className="recipe-description">
            { recipe.description }
          </div>
          { recipe.instructions && (
          <div className="recipe-steps">
            <h3>Instructions</h3>
            <ul>
              {
                recipe.instructions.map((instruction) => (
                  <li key={instruction.id}>
                    {instruction.display_text}
                  </li>
                ))
                }
            </ul>
          </div>
          )}
        </div>
      ) }
    </div>
  );
};

export default Recipe;
