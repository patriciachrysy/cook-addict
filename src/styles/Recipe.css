@import url('https://fonts.googleapis.com/css2?family=Kanit:wght@500&family=Pacifico&family=Rubik:wght@300&display=swap');

.recipe-details {
  height: 100vh;
  overflow-y: scroll;
}

.recipe-banner {
  width: 100%;
}

.recipe-banner img {
  width: 100%;
  height: 35vh;
}

.recipe-banner video {
  width: 100%;
}

.recipe-infos {
  padding: 10px;
}

.recipe-infos h1 {
  font-family: 'Kanit', sans-serif;
  font-size: 40px;
  margin: 0;
}

.metrics {
  display: flex;
  justify-content: space-between;
  padding: 10px 0;
}

.metrics span {
  display: flex;
  align-items: center;
  font-size: 15px;
  font-weight: bold;
}

.metrics span svg {
  font-size: 20px;
  margin: 0 5px;
}

.tags {
  display: flex;
  flex-wrap: wrap;
}

.tags span {
  font-size: 10px;
  margin: 3px;
  padding: 2px;
  border: 1px solid black;
  border-radius: 5px;
}

.recipe-description {
  padding: 10px;
  font-family: 'Kanit', sans-serif;
}

.recipe-steps {
  padding: 5px 20px;
}

.recipe-steps h3 {
  font-family: 'Kanit', sans-serif;
  text-decoration: underline;
}

.recipe-steps ul {
  font-family: 'Rubik', sans-serif;
}

.recipe-steps ul li {
  margin: 10px 0;
}
