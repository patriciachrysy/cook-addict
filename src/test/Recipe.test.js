import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Recipe from '../components/Recipe';

import store from '../Redux/ConfigureStore';


it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Router><Recipe /></Router></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Recipe component', () => {
  render(<Provider store={store}><Router><Recipe /></Router></Provider>);
  screen.debug();
});
