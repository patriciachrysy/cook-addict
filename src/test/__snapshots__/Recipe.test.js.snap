// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="spinner-container"
  >
    <div
      className="loading-spinner"
    />
  </div>
</div>
`;
